Xwt.XamMac/Xwt.Mac/NSApplicationInitializer.cs

```diff
@@ -24,23 +24,22 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+using System;
+
 using AppKit;
 
 namespace Xwt.Mac
 {
 	static class NSApplicationInitializer
 	{
+		[ThreadStatic]
+		static bool initialized;
```

Comment:
I don't get why we're using a thread-local storage for this. Shouldn't the app only be initialized on the main thread?

One could potentially call NSApplication.Init on 2 different threads. This seems wrong.

Reply:
Actually calling initialize multiple times makes it throw:
https://github.com/xamarin/xamarin-macios/blob/f1cb31004382bfe2bf87ae0f047534273d06672c/src/AppKit/NSApplication.cs#L52

Reply:
I did not want to change the semantics of anything. I agree with you though - I do not think you should ever be allowed to call this from multiple threads, but I was going for a perf gain over changing any semantics.

Reply:
Okay, I kind-of understand why it's done like this. Multiple inits from the main thread would not throw.

Reply:
Yeah, I think the desire was to allow multiple init calls from multiple threads, but allow `NSApplication.Init` itself to enforce multiple-inits _per thread_ by throwing. But again, I don't have the back story as to why it was done like this. I just wanted to modernize the threading semantics in place. /cc @slluis.